doc/design/drafts/dynamic_size_types.md

```diff
@@ -0,0 +1,562 @@
+# Dynamic Size Support for Zero Copy Types
+
+## Summary
+
+There are multiple obstacles when it comes to use true zero copy with types
+whose size is not known at compile time. At the moment a developer can pursue
+three approaches.
+
+1. Send via untyped publisher and provide the size at runtime to the publisher.
+   But defining message types with multiple dynamic sized containers would be
+   a real challenge when they should be constructed inside of this sample.
+
+2. One can perform a worst case estimation at compile time and declare
+   a type with container members which fullfil those estimations.
+   This could lead to a lot of wasted memory and if the worst case estimations
+   have to be adjusted all the applications which are using this type may have
+   to be adjusted as well since the capacity is part of the type.
+
+3. Another approach is to use the heap and copy/serialize the data into a sample
+   before sending it with iceoryx. This would imply multiple copies since the
```

Comment:
Leads to one serialization and one deserialization and is optimal *if data has to be serialized anyway*.

No copies needed if done correctly, but tota effort is effort of Omega(size of data to be sent), i.e. at least linear as a copy with more complex logic (leading to increased costs compared to copies).